cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/TopicContentsService.java

```diff
@@ -107,11 +119,19 @@ public Map<Long, String> readEvents(
     }
 
     int i = 0;
+    boolean exitLoop = false;
     do {
       ConsumerRecords<String, String> consumerRecords = consumer.poll(Duration.ofMillis(500));
-      consumerRecords.forEach(record -> eventMap.put(record.offset(), record.value()));
+      for (ConsumerRecord<String, String> record : consumerRecords) {
+        eventMap.put(record.offset(), record.value());
+        if (offsetPosition.equals(RANGE_OFFSET_SELECTION)
+            && (record.offset() >= rangeOffsetsEnd || eventMap.size() >= RANGE_MAX_RECORDS)) {
```

Comment:
don't we need range offsets start position ?

Reply:
I followed the existing pattern and used the `rangeOffsetsStart` [here](https://github.com/khatibtamal/klaw/blob/1987-view-topic-events-range-backend/cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/TopicContentsService.java#L109) as we seek to the start position [here](https://github.com/khatibtamal/klaw/blob/1987-view-topic-events-range-backend/cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/TopicContentsService.java#L117).

Inside the loop we check the `rangeOffsetsEnd` only if the selection is **range** this ensures that the existing default behavior is completely unchanged.

Reply:
looks good